packages/sit-onyx/src/composables/useAnchorPositionPolyfill.ts

```diff
@@ -27,11 +30,62 @@ export const useAnchorPositionPolyfill = ({
   const leftPosition = ref("-1000px");
   const topPosition = ref("-1000px");
 
+  let targetVisible = false;
+  const scrollParent = computed(() => getScrollParent(getElement(targetRef)));
+
+  const targetObserver = new IntersectionObserver((entries) => {
+    entries.forEach((entry) => {
+      targetVisible = entry.isIntersecting;
+      updateAnchorPositionPolyfill();
+      if (targetVisible) {
+        const positionedElement = getElement(positionedRef);
+        if (positionedElement) {
+          positionedObserver.observe(positionedElement);
+        }
+      } else {
+        positionedObserver.disconnect();
+        scrollParent.value?.removeEventListener("scroll", updateAnchorPositionPolyfill);
+
+        leftPosition.value = "-1000px";
+        topPosition.value = "-1000px";
+      }
+    });
+  });
+
+  const positionedObserver = new IntersectionObserver((entries) => {
+    entries.forEach((entry) => {
+      if (entry.isIntersecting) {
+        scrollParent.value?.addEventListener("scroll", updateAnchorPositionPolyfill);
+      } else {
+        scrollParent.value?.removeEventListener("scroll", updateAnchorPositionPolyfill);
+        leftPosition.value = "-1000px";
+        topPosition.value = "-1000px";
+      }
+    });
+  });
+
   const getElement = (refOrGetter: MaybeRefOrGetter<HTMLElement | null>): HTMLElement | null => {
     const element = unref(refOrGetter);
```

Comment:
Use [`toValue`](https://vuejs.org/api/reactivity-utilities.html#tovalue) instead of `unref` for resolving `MaybeRefOrGetter` types. This will also handle getters.
```suggestion
    const element = toValue(refOrGetter);
```